downloader/src/main/kotlin/WorkingTreeCache.kt

```diff
@@ -55,11 +56,23 @@ interface WorkingTreeCache {
 
 class DefaultWorkingTreeCache : WorkingTreeCache {
     private val mutex = Mutex()
+    private val vcsPluginConfigs = mutableMapOf<String, PluginConfig>()
     private val workingTreeMutexes = mutableMapOf<String, Mutex>()
     private val workingTrees = mutableMapOf<String, WorkingTree>()
 
     private var terminated = false
 
+    /**
+     * Add [VCS plugin configurations][configs] that are applied when creating a [VersionControlSystem] instance
+     * for this working tree. Calling this function is optional.
+     */
+    @Suppress("unused") // Called externally when used as a library
```

Comment:
Comment nits: Generally, cannot know *if* the function is called externally. We just provide it "in case someone needs it". So I propose something like "This is intended to be used programmatically by third parties." (also adding a dot at the end of the sentence). But feel free to come up with something less "bulky".